### ISSUE TITLE
Release tool improvements

### ISSUE THREAD

**Issue by nawazkh** [kubernetes-sigs/cluster-api#9008]
This issue aims to track the improvements needed/done at [release tool](https://github.com/kubernetes-sigs/cluster-api/blob/main/hack/tools/release/notes.go)
Assuming this to be a long-term effort, this issue will be used as an epic and subsequent tasks will be created to track the changes.

```[tasklist]
### Release Tool Improvements
- [ ] Draft task: fix up grammatically incorrect plurals in the release tool i.e. `1 new commits merged`
```

**Comment by killianmuldoon** [comment]
/triage accepted 

I don't think there's any reason to open up seperate issues for each of these as they're quite small and multiple might be done in a single PR.

**Comment by nawazkh** [comment]
> /triage accepted
> 
> I don't think there's any reason to open up separate issues for each of these as they're quite small and multiple might be done in a single PR.

Oops, sorry! I created a few already; did not see your reply here.
Feel free to manage the other issues, if you would like to close them out or keep it as is. :)

**Comment by g-gaston** [comment]
@nawazkh can we add another one to the list?

> When there are multiple areas, areas should be replaces with a well formatted version of their current rendering
> clustercachetracker|provider/control-plane-kubeadm|provider/bootstrap-kubeadm
> To
> ClusterCacheTracker/KCP/CAPBK.

**Comment by killianmuldoon** [comment]
>  When there are multiple areas, areas should be replaces with a well formatted version of their current rendering
    clustercachetracker|provider/control-plane-kubeadm|provider/bootstrap-kubeadm

I think we have to be careful with this as these probably do need to be manually reviewed to be sure there aren't issues with the multiple areas added. Maybe we could add another marker to signal that multiple areas are okay for a given PR so we don't have to manually fix it up each time we generate the notes.

**Comment by g-gaston** [comment]
> I think we have to be careful with this as these probably do need to be manually reviewed to be sure there aren't issues with the multiple areas added.

Yeah, for know I'm suggesting just making sure we do the `provider/control-plane-kubeadm` -> `KCP` replacement
Same overall format with `MULTIPLE_AREAS[*]`. So it will still need review but at least it's just about deciding what to keep and what don't, but the areas themselves are user friendly.

> Maybe we could add another marker to signal that multiple areas are okay for a given PR so we don't have to manually fix it up each time we generate the notes.

I would suggest tackling this as a separate task. It seems we might need to introduce a new mechanism, so it's not as cheap as just doing the replacement and might need a bit more thought.

**Comment by killianmuldoon** [comment]
>Yeah, for know I'm suggesting just making sure we do the provider/control-plane-kubeadm -> KCP replacement
Same overall format with MULTIPLE_AREAS[*]

That sounds perfect! Let me know if the task I added above captured what you want. 

 > Maybe we can revisit based on how frequent these are? Maybe they are rare and it's not worth the added complexity.

Completely agreed - I'll add a generic task for this on this umbrella issue, but I think you're right that it's not very high priority.

**Comment by sbueringer** [comment]
Just posting here because it's related. Going forward let's please consider that the release note tool can also be used by non-core CAPI. I.e. we should be careful about hard-coding the cluster-api repo name or adding features (without flags) that don't make sense for other consumers (xref: https://github.com/kubernetes-sigs/cluster-api/pull/9018#issuecomment-1643850235)

**Comment by g-gaston** [comment]
> Just posting here because it's related. Going forward let's please consider that the release note tool can also be used by non-core CAPI. I.e. we should be careful about hard-coding the cluster-api repo name or adding features (without flags) that don't make sense for other consumers (xref: [#9018 (comment)](https://github.com/kubernetes-sigs/cluster-api/pull/9018#issuecomment-1643850235))
> 
> P.S. the area mapping is totally fine

Should we add a README to the folder stating this? Or any other way to document this goal that you see fit.

To be honest, I didn't know this was a goal of the tool. So I fear if we don't document it, someone else might come in the future without this background and make breaking changes again.

**Comment by killianmuldoon** [comment]
Let's also add this as a comment on `main.go` in the release notes tool as an additional hint as maintenance of this tool is likely to swap hands frequently with release team handovers. 

I also wasn't aware that we were considering non-CAPI users of the tool, but it makes complete sense for providers at least.

**Comment by sbueringer** [comment]
Just to share data. Currently used by 4 other providers (soon 5 with CAPV): https://cs.k8s.io/?q=sigs.k8s.io%2Fcluster-api%2Fhack%2Ftools%2Frelease&i=nope&files=&excludeFiles=&repos=

**Comment by killianmuldoon** [comment]
@kubernetes-sigs/cluster-api-release-team to make sure I captured all the issues we spoke about on today's call.
